### PR TITLE
Add workspace 

### PR DESCRIPTION
**Pull request by pysel**
## 1. Summary

This PR adds a Go workspace to Quicksilver, ensuring that the code is easier to manage holistically. 

## 2.Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 3. Implementation details

Just adds a workspace, as the description says. 

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: test/interchaintest/go.mod [hunk 1]
```diff
@@ -211,6 +211,6 @@ replace (
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 	github.com/ingenuity-build => ../../
 	github.com/strangelove-ventures/interchaintest/v5 => github.com/aljo242/interchaintest/v5 v5.0.0-20230619200058-1d4823df8b75
-	github.com/tendermint/tendermint => github.com/cometbft/cometbft v0.34.27
+	github.com/tendermint/tendermint => github.com/cometbft/cometbft v0.34.32
```
Comment (pysel) [inline test/interchaintest/go.mod:214]: this is needed because this version conflicts with other modules' version of comet

**Comment by faddat** [comment]
re: the workspace -- happy to accept this, even, I'd like it, but in the past I had trouble getting it to work.

In fact, in general, I've had pretty widespread issues with go workspaces.

(but if you are able to fix the build, and maybe comment some on how to use the workspace, then I'd very much like this PR)

**Comment by pysel** [comment]
hey @faddat, should be good now, could you please enable the ci on this pr please to check it? locally, the build works just fine!

**Review by tropicaldog — APPROVED** [review]

**Comment by faddat** [comment]
I'm so in.

Thank you!

**Review by faddat — APPROVED** [review]

**Review by faddat — APPROVED** [review]
100% yes

**Pull request merged**